src/codegen_call_thread.cpp

```
@@ -92,474 +92,418 @@ static std::unordered_set<std::string> collectReferencedVars(const LambdaExpr &l
-            codegenError("thread_join() requires Thread argument");
-        auto fn = getRuntimeFn("__ry_thread_join", i64Ty_, {ptrTy_});
-        llvm::Value *status = builder_.CreateCall(fn, {thread}, "join_status");
-        // Don't nullify: __ry_thread_join only joins (no free). ARC cleanup at
-        // scope exit will call __ry_thread_cleanup which sees joinable()==false
-        // and just destructs the handle, then ARC frees the block.
-        return wrapStatusAsResult(status);
-    }
+    // Call __ry_thread_spawn(thunk, env)
+    auto spawnTy = llvm::FunctionType::get(ptrTy_, {ptrTy_, ptrTy_}, false);
+    auto spawnFn = mod_->getOrInsertFunction("__ry_thread_spawn", spawnTy);
+    llvm::Value *thread = builder_.CreateCall(
```

Clang (14.0.6)

[warning] variable 'thread' is not initialized

(cppcoreguidelines-init-variables)